### PR TITLE
Initial support for GCP deployments

### PR DESCRIPTION
**Pull request by shadaj**
Also switches over some parts to use anyhow for error reporting, although I will do more refactoring for this in a later PR.

### CODE REVIEW

File: hydro_cli/src/core/hydroflow_crate.rs [hunk 1]
```diff
@@ -178,22 +178,22 @@ impl Service for HydroflowCrate {
...
-            .await;
+            .launch_binary(self.built_binary.as_ref().unwrap())
+            .await?;
 
         let mut bind_types = HashMap::new();
         for (port_name, from_host) in self.incoming_ports.iter() {
             bind_types.insert(
                 port_name.clone(),
-                from_host
+                self.on
```
Comment (shadaj) [inline hydro_cli/src/core/hydroflow_crate.rs:193]: oops, annoying bug that only shows up when you actually have two hosts

File: hydro_cli/src/core/gcp.rs [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+    for _ in 1..count {
+        let result = thunk().await;
+        if result.is_ok() {
+            return result;
+        } else {
+            tokio::time::sleep(std::time::Duration::from_millis(1000)).await;
+        }
+    }
+
+    thunk().await
```
Comment (MingweiSamuel) [inline hydro_cli/src/core/gcp.rs:79]: Any reason for not just doing this as sequential async code?

File: hydro_cli/src/core/gcp.rs [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+        let stderr_receivers = create_broadcast(channel.stderr(), |s| eprintln!("{s}"));
+
+        Ok(Arc::new(RwLock::new(LaunchedComputeEngineBinary {
+            _resource_result: self.resource_result.clone(),
+            channel,
+            stdin_channel: stdin_sender,
+            stdout_receivers,
+            stderr_receivers,
+        })))
+    }
```
Comment (MingweiSamuel) [inline hydro_cli/src/core/gcp.rs:162]: More comments thruout would be good

**Review by MingweiSamuel — APPROVED** [review]

File: hydro_cli/src/core/gcp.rs (hunk 2 above)
Comment (shadaj) [reply hydro_cli/src/core/gcp.rs:79]: Not sure I understand your question. The reason for the extra call at the end is so we can propagate errors back up to the user.

**Pull request merged**